apps/api/src/persistence/repositories/arango-id-repository.ts

```diff
@@ -1,50 +1,83 @@
+import { UniquelyIdentifiableType } from '../../domain/interfaces/id-manager.interface';
 import { AggregateId } from '../../domain/types/AggregateId';
+import { isNullOrUndefined } from '../../domain/utilities/validation/is-null-or-undefined';
 import { InternalError } from '../../lib/errors/InternalError';
 import { IIdRepository } from '../../lib/id-generation/interfaces/id-repository.interface';
 import { UuidDocument } from '../../lib/id-generation/types/UuidDocument';
 import { Maybe } from '../../lib/types/maybe';
 import { isNotFound, NotFound } from '../../lib/types/not-found';
 import { ArangoDatabase } from '../database/arango-database';
 import { ArangoCollectionId } from '../database/collection-references/ArangoCollectionId';
-import { DatabaseProvider } from '../database/database.provider';
-import mapDatabaseDTOToEntityDTO from '../database/utilities/mapDatabaseDocumentToAggregateDTO';
-import mapEntityDTOToDatabaseDTO, {
-    DatabaseDocument,
-} from '../database/utilities/mapEntityDTOToDatabaseDTO';
+import { ArangoDatabaseProvider } from '../database/database.provider';
 
-type IdDocument = DatabaseDocument<{ id: AggregateId; isAvailable: boolean }>;
+type DatabaseUuidDocument = Omit<UuidDocument, 'sequenceNumber' | 'id'> & {
+    _key: string;
+    uuid: string;
+};
 
-export class ArangoIdRepository implements IIdRepository<AggregateId> {
+const mapDatabaseDocumentToUuidDocument = ({
```

Comment:
Can't we use the 'cloneWithoutProperties' helper to make this a little less tedious?